### ISSUE TITLE
mpmath.fp.diff default method not working

### ISSUE THREAD

**Issue by DerWeh** [fredrik-johansson/mpmath#474]
Minimal example:

```python3
import mpmath
import numpy as np
print(mpmath.__version__)
print(mpmath.fp.diff(np.exp, 0))
print(mpmath.fp.diff(np.exp, 0, method='quad'))

# output:
# 1.1.0
# 0.0
# (0.9999999999999988+1.3912230833030279e-16j)
```

The method `'quad'` works, while the default methods silently gives nonsense.

**Comment by fredrik-johansson** [comment]
Yes, this really needs to be fixed. However, rather than making 'quad' the default, the default for fp.diff should be an algorithm using high-order stencils.

https://en.wikipedia.org/wiki/Finite_difference_coefficient

For example, the following gives nearly full accuracy:

    >>> C = [1/280., -4/105., 1/5., -4/5., 0.0, 4/5., -1/5., 4/105., -1/280.]
    >>> x=0.0; h=0.02; sum(C[i]*fp.exp(x+h*(i-4)) for i in range(9))/h;
    0.9999999999999997
